### ISSUE TITLE
BuildInfo isEnterprise() does not work in 3.7.x

### ISSUE THREAD

**Issue by martinfurmanski** [hazelcast/hazelcast#10740]
Code:
```
com.hazelcast.instance.BuildInfoProvider.getBuildInfo().isEnterprise()
```

With:

```
<dependency>
    <groupId>com.hazelcast</groupId>
    <artifactId>hazelcast-enterprise-all</artifactId>
    <version>3.7.8</version>
</dependency>
```

Expected: true
Returns: false

tested 3.7.5 and 3.7.8: broken
tested 3.8.2: works fine

I did some investigations and the hazelcast-runtime.properties resource embedded in the enterprise distribution simply has:

```
hazelcast.version = 3.7.5
hazelcast.build = 20170124
hazelcast.distribution = Hazelcast
hazelcast.git.revision = 111f332
hazelcast.serialization.version = 1
```

which seems to be the root of the problem.

**Comment by martinfurmanski** [comment]
The reason I need this is because HZ will silently fail and revert to non-secure communication even if I configure/enable SSL but happen to run with the wrong dependencies. This is by itself quite bad, but being able to detect Enterprise at least gives me a chance to fail rather than continue on in a non-secure way.

**Comment by Donnerbart** [comment]
The `BuildInfo` generation was refactored in Hazelcast 3.9, see https://github.com/hazelcast/hazelcast/pull/10139

There is also a check if security is enabled and Hazelcast is not running with EE now: https://github.com/hazelcast/hazelcast/pull/11143

**Comment by martinfurmanski** [comment]
@Donnerbart fix in 3.7? it is broken there

**Comment by Donnerbart** [comment]
I don't think that we can backport all things to the 3.7.x line, our current maintenance version is 3.8.x. Since the BuildInfo refactoring was part of a big PR, it's best to ask @jerrinot about this. The Security check should be easier to backport @pveentjer.
